src/ucp/wireup/wireup_ep.c

```diff
@@ -331,33 +331,50 @@ static ucs_status_t ucp_wireup_ep_flush(uct_ep_h uct_ep, unsigned flags,
     return UCS_ERR_NO_RESOURCE;
 }
 
+static ucs_status_t
+ucp_wireup_ep_do_check(ucp_ep_h checked_ucp_ep, ucp_ep_h ucp_ep,
+                       uct_ep_h uct_ep, ucp_rsc_index_t rsc_idx, unsigned flags,
+                       uct_completion_t *comp, int strict_ep_check)
+{
+    ucp_worker_h worker = checked_ucp_ep->worker;
+    ucp_worker_iface_t *wiface;
+
+    ucs_assert(rsc_idx != UCP_NULL_RESOURCE);
+    ucs_assert(ucp_ep_remote_id(checked_ucp_ep) == ucp_ep_remote_id(ucp_ep));
+
+    wiface = ucp_worker_iface(worker, rsc_idx);
+    ucs_assert(!strict_ep_check ||
+               (wiface->attr.cap.flags & UCT_IFACE_FLAG_EP_CHECK));
```

Comment:
do we require this `UCT_IFACE_FLAG_EP_CHECK` flag to do new wireup based KA?
imo, we do not have to do that

Reply:
> do we require this `UCT_IFACE_FLAG_EP_CHECK` flag to do new wireup based KA?

if a lane is in ep_check_map, it means that this has `UCT_IFACE_FLAG_EP_CHECK` flag

Reply:
why do we need to require `UCT_IFACE_FLAG_EP_CHECK` for UCT eps where we do AM/wireup based KA?
Also can it be that the lane supports `UCT_IFACE_FLAG_EP_CHECK` and `UCT_IFACE_FLAG_CONNECT_TO_IFACE`, but does not support AM?

Reply:
>  why do we need to require UCT_IFACE_FLAG_EP_CHECK for UCT eps where we do AM/wireup based KA?

The idea is to do AM/wireup for EP which supports EP check and uses connect_to_idwce method. It is the same as KA for main EPs.

If KA is enabled, it means that error handling is requested and AM is required. So, AM will be sent through either wireup lane or AM lane.